plexpy/api.py

```diff
@@ -217,9 +289,211 @@ def _checkGithub(self, **kwargs):
 
     def _shutdown(self, **kwargs):
         plexpy.SIGNAL = 'shutdown'
+        self.msg = 'Shutting down plexpy'
+        self.result_type = 'success'
 
     def _restart(self, **kwargs):
         plexpy.SIGNAL = 'restart'
+        self.msg = 'Restarting plexpy'
+        self.result_type = 'success'
 
     def _update(self, **kwargs):
         plexpy.SIGNAL = 'update'
+        self.msg = 'Updating plexpy'
+        self.result_type = 'success'
+
+    def _getHistory(self, user=None, user_id=None, rating_key='', parent_rating_key='', grandparent_rating_key='', start_date='', **kwargs):
+
+        custom_where = []
+        if user_id:
+            custom_where = [['user_id', user_id]]
+        elif user:
+            custom_where = [['user', user]]
+        if 'rating_key' in kwargs:
+            rating_key = kwargs.get('rating_key', "")
+            custom_where = [['rating_key', rating_key]]
+        if 'parent_rating_key' in kwargs:
+            rating_key = kwargs.get('parent_rating_key', "")
+            custom_where = [['parent_rating_key', rating_key]]
+        if 'grandparent_rating_key' in kwargs:
+            rating_key = kwargs.get('grandparent_rating_key', "")
+            custom_where = [['grandparent_rating_key', rating_key]]
+        if 'start_date' in kwargs:
+            start_date = kwargs.get('start_date', "")
+            custom_where = [['strftime("%Y-%m-%d", datetime(date, "unixepoch", "localtime"))', start_date]]
+
+        data_factory = datafactory.DataFactory()
+        history = data_factory.get_history(kwargs=kwargs, custom_where=custom_where)
+
+        self.data = history
+        return self.data
+
+    def _getSync(self, machine_id=None, user_id=None, **kwargs):
```

Comment:
Nope, its done, but i didnt notice that one, must have missed it while i copy pasted